trace/trace.go

```diff
@@ -57,36 +57,39 @@ func (c *TraceableContext) Pop() context.Context {
 
 // Start starts a span and return a finish() function to finish the corresponding span.
 func Start(c context.Context, operation string, spanOpts ...sentry.SpanOption) func() {
-	if !viper.GetBool("sentry.on") {
+	if viper.GetFloat64("sentry.tracesSampleRate") == 0 { // if trace sample rate is 0.0 or 0
 		return func() {}
-	}
+	} else if !viper.GetBool("sentry.on") { // if sentry is off
+		return func() {}
+	} else {
 
```

Comment:
too many `if else`, I think this is enough
```
if !viper.GetBool("sentry. On") || viper.GetFloat64("sentry.tracesSampleRate") == 0 {
    return func() {}
}
```

Reply:
Thanks. will do